### PR TITLE
Add generic MCP server registration via config file

### LINKED ISSUE #68: Evaluate optional generic MCP stdio registration in bot config
## Summary
Track optional support for registering arbitrary MCP stdio servers from bot configuration/env.

This is intentionally separated from the internal `telegram-channel` auto-registration work to keep scope small and delivery simple.

## Why
- Some deployments may want to expose extra MCP servers without changing code.
- Useful for experimentation and local integrations.

## Scope ideas
- Optional env/CLI config for one or more MCP stdio servers.
- Validation and safe parsing of command/args/env.
- Clear precedence between internal MCP server and external configured servers.

## Out of scope
- Core Telegram channel tools implementation.
- Session/chat routing logic for channel side effects.

## Acceptance criteria
- Config format documented.
- Tests for parsing and forwarding to ACP `new_session` / `load_session`.
- No regression in default no-config behavior.

### PR DESCRIPTION
**Pull request by mgaitan**
## Summary

- `mcp_servers` key in the JSON config file registers external MCP servers
- Supports both **stdio servers** (local subprocess via `command`) and **remote servers** (HTTP via `url`)
- Internal `telegram-channel` server is always included; a configured server named `telegram-channel` overrides it
- `AcpAgentService` now accepts `McpServerStdio | HttpMcpServer` tuples
- Startup validation with clear error messages for invalid server definitions

Closes #68

**Depends on #192** (unified config system — this PR is stacked on `feat/191-unified-config`)

## Config examples

```json
{
  "mcp_servers": {
    "filesystem": {
      "command": "npx",
      "args": ["@modelcontextprotocol/server-filesystem", "/home/alice/projects"]
    },
    "remote": {
      "url": "https://mcp.example.com/mcp",
      "headers": { "Authorization": "Bearer token" }
    }
  }
}
```

## Changes

- `src/telegram_acp_bot/config_file.py` — adds `_validate_mcp_servers_section`, `_validate_mcp_server_entry`, `_validate_stdio_server`, `_validate_http_server`
- `src/telegram_acp_bot/__init__.py` — `_default_mcp_servers()` now accepts `extra_servers` dict and merges them
- `src/telegram_acp_bot/acp/service.py` — `mcp_servers` type widened to `tuple[McpServerStdio | HttpMcpServer, ...]`
- `tests/test_config_file.py` — validation tests for all mcp_servers edge cases
- `tests/test_cli.py` — integration tests for stdio, HTTP, and conflict-resolution scenarios
- `docs/mcp.md` — new "External MCP Servers" section with examples and troubleshooting

## Test plan

- [x] 527 tests pass, 100% coverage
- [x] ruff lint clean, ty type-check clean
- [x] Test: stdio server passed to service with correct command/args/env
- [x] Test: HTTP server passed to service with correct url/headers
- [x] Test: named conflict — external `telegram-channel` overrides internal
- [x] Test: all validation error paths (missing transport, both transports, wrong types)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by mgaitan** [comment]
@claude   names should match

 (^[a-z0-9-_]+$)  and no spaces

This gonna simplify logs, debugging and UX.

File: src/telegram_acp_bot/__init__.py [hunk 1]
```diff
@@ -147,21 +147,42 @@ def _default_mcp_servers(
...
-            ],
-        ),
+    extra_servers: dict[str, Any] | None = None,
+) -> tuple[McpServerStdio | HttpMcpServer, ...]:
+    """Return MCP servers to expose to the ACP agent.
+
+    Always includes the internal ``telegram-channel`` server.
+    External servers from the config file are appended after it.
+    If an external server name conflicts with the internal one, the external
+    server overrides it.
```
[the remark this answers is not in the document]
Comment (mgaitan) [reply src/telegram_acp_bot/__init__.py]: Applied in 87791ec. The docstring now matches the actual ordering semantics when a configured server overrides `telegram-channel`.

**Comment by mgaitan** [comment]
Addressed the remaining review feedback in `87791ec`.

- MCP server keys now must match `^[a-z0-9-_]+$` (lowercase, no spaces), with validation coverage.
- The `_default_mcp_servers()` docstring now matches the real override ordering for `telegram-channel`.
- I also fixed a stale `set_config_option()` type override in the demo agent so fresh `ty check` passes in a clean worktree.

Local validation:
- `uv run pytest`
- `uv run ty check`
